firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/SetMutation.java

```diff
@@ -72,7 +72,8 @@ public MaybeDocument applyToRemoteDocument(
     // accepted the mutation so the precondition must have held.
 
     SnapshotVersion version = mutationResult.getVersion();
-    return new Document(getKey(), version, value, Document.DocumentState.COMMITTED_MUTATIONS);
+    return Document.fromObjectValue(
```

Comment:
The SetMutation `value` field could be similarly constructed from a proto such that this could use the `fromProto` version. It's possible this is something for a follow-up PR.

Reply:
Yeah, I want to address the mutations afterwards.